### PR TITLE
Limit the number of requests popped from pending set

### PR DESCRIPTION
**Pull request by justinvdm**
See [here](https://github.com/praekelt/vumi-http-retry-api/pull/6#discussion_r35755407) for context.

### CODE REVIEW

**Comment by justinvdm** [comment]
Ready for review.

File: vumi_http_retry/retries.py [hunk 1]
```diff
@@ -53,15 +53,35 @@ def add_pending(redis, prefix, req):
...
     returnValue([json.loads(r) for r in reqs])
 
 
 @inlineCallbacks
+def pop_pending(redis, prefix, from_time, to_time, chunk_size=500):
+    results = []
+
+    while True:
+        reqs = yield limit_pop_pending(
+            redis, prefix, from_time, to_time, chunk_size)
```
Comment (hodgestar) [inline vumi_http_retry/retries.py]: I think this loop should happen where we call `add_ready` so we aren't trying to add an unlimited number of reqs into the ready set at once either?

Comment (3leanore) [inline vumi_http_retry/retries.py]: Ya, didn't think this one through. Will get this working in a way that we aren't popping or adding an unlimited number of requests, and without keeping everything in memory halfway through.

Comment (justinvdm) [inline vumi_http_retry/retries.py]: Ya, didn't think this one through. Will get this working in a way that we aren't popping or adding an unlimited number of requests, and without keeping everything in memory halfway through.

**Comment by justinvdm** [comment]
Ready for re-review.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**